os/drivers/os_api_test/application/test_app.c

```diff
@@ -0,0 +1,91 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <errno.h>
+
+#include <tinyara/mpu.h>
+#include <tinyara/os_api_test_drv.h>
+
+#ifdef CONFIG_EXAMPLES_DAT
+#ifdef CONFIG_SYSTIME_TIME64
+extern uint64_t app_start_time_sec[3];
+#else
+extern uint32_t app_start_time_sec[3];
+#endif
+extern unsigned int app_start_time_csec[3];
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/************************************************************************************
+ * Name: test_get_app_start_time
+ *
+ * Description:  This function retrieves the application start time
+ *
+ ************************************************************************************/
+
+#ifdef CONFIG_SYSTIME_TIME64
+uint64_t test_get_app_start_time_sec(int cmd, unsigned long arg)
+#else
+uint32_t test_get_app_start_time_sec(int cmd, unsigned long arg)
+#endif
+{
+#ifdef CONFIG_SYSTIME_TIME64
+	uint64_t sec = -EINVAL;
+#else
+	uint32_t sec = -EINVAL;
+#endif
+
+	switch (cmd) {
```

Comment:
Please see if we need to add CONFIG_EXAMPLES_DAT check here as app_start_time_sec definition has a dependency on the config.